### ISSUE TITLE
performance: forced-reflow in simplebar constructor

### ISSUE THREAD

**Issue by staabm** [Grsmto/simplebar#162]
when calling the SimpleBar constructor on a simple div, it triggers forced-reflows which slows down rendering performance of the web-browsers.

```js
      var lDiv = parentNode.querySelector('.custom-select-l');
      lDiv.innerHTML = html; // my html
      new SimpleBar(lDiv);
```

I can see serveral reflows with the chrome performance panel:

![grafik](https://user-images.githubusercontent.com/120441/38247960-320ad4e0-3747-11e8-9049-00584f39ac06.png)

please find the exported chrome profile, for better in-deep analysis:
[simplebar-profile.zip](https://github.com/Grsmto/simplebar/files/1871370/simplebar-profile.zip)

see https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing#avoid-forced-synchronous-layouts for further background.

the source in the profile references simplebar.js minified sources which might make debugging harder. Since simplebar does not provide a non minified version, reporting a good bug report is not so easy, see https://github.com/Grsmto/simplebar/issues/161.

_simplebar version_:   v2.6.1
_Browser name && version_:  Chrome 67.0.3387.0
_OS name && version_:   Windows10 x64

**Comment by staabm** [comment]
pretty-printing the minified sources with chrome results in the following offending lines:

![grafik](https://user-images.githubusercontent.com/120441/38248324-4506bfae-3748-11e8-8aa1-8898b856f25a.png)

![grafik](https://user-images.githubusercontent.com/120441/38248346-536e5426-3748-11e8-9e8b-ac1d9e5a8d2c.png)

**Comment by Grsmto** [comment]
Hi,
Thanks for this very detailed report.
SimpleBar is doing loads of read/write to the DOM so I'm not surprised this is happening. The call to `getComputedStyle` has been added recently and I believe this is a fairly slow function. I think it would be better to get rid of it entirely (using an option to activate the RTL seems more reliable).

About the other ones, maybe the `recalculate` should be wrapped in a RAF or use something like fastDOM (or just kind of do every DOM r/w as bulk operations).

**Comment by staabm** [comment]
Making rtl an option instead of auto detectin it would really be usefull for most cases.

Also separating read write is also the way to go. Thank you

**Comment by Grsmto** [comment]
Closed in simplebar@3.0.0-beta.0.

**Comment by staabm** [comment]
thank you for working on it.

it tried the latest 3.0.0-beta1 and still can see forced-reflows while rendering:

![grafik](https://user-images.githubusercontent.com/120441/42747438-a0538668-88dc-11e8-8596-5624b4f97c63.png)

the recalculate style and the reflow seem to be triggered on the same JS line

recalculate style (in scrollbarWidth())
![grafik](https://user-images.githubusercontent.com/120441/42747444-ab63b5e6-88dc-11e8-907f-4ebdfab0a451.png)

reflow (in scrollbarWidth())
![grafik](https://user-images.githubusercontent.com/120441/42747473-cd00cb94-88dc-11e8-9eb6-70265caada90.png)

the 2nd recalculate style which happens during render() is triggered here:
![grafik](https://user-images.githubusercontent.com/120441/42747500-e24062b2-88dc-11e8-8eef-baf91f12b59f.png)
